l5kit/l5kit/simulation/unroll.py

```diff
@@ -0,0 +1,226 @@
+from typing import Dict, List, NamedTuple, Optional, Tuple
+
+import numpy as np
+import torch
+from torch.utils.data.dataloader import default_collate
+from tqdm.auto import tqdm
+
+from l5kit.data import AGENT_DTYPE
+from l5kit.dataset import EgoDataset
+from l5kit.geometry import transform_points
+from l5kit.simulation.dataset import SimulationDataset
+
+
+class SimulationConfig(NamedTuple):
+    """ Defines the parameters used for the simulation of ego and agents around it.
+    :param use_ego_gt: whether to use GT annotations for ego instead of model's outputs
+    :param use_agents_gt: whether to use GT annotations for agents instead of model's outputs
+    :param disable_new_agents: whether to disable agents that are not returned at start_frame_index
+    :param distance_th_far: if a tracked agent is closed than this value to ego, it will be controlled
+    :param distance_th_close: if a new agent is closer than this value to ego, it will be controlled
+    :param start_frame_index: the start index of the simulation
+    :param num_simulation_steps: the number of step to simulate
+    """
+    use_ego_gt: bool
+    use_agents_gt: bool
+    disable_new_agents: bool
+    distance_th_far: float
+    distance_th_close: float
+    start_frame_index: int = 0
+    num_simulation_steps: Optional[int] = None
+
+
+class SimulationOutputs:
+    def __init__(self, scene_id: int, sim_dataset: SimulationDataset):
+        """This object holds information about the result of the simulation loop
+        for a given scene dataset
+
+        :param scene_id: the scene indices
+        :param sim_dataset: the simulation dataset
+        """
+        if scene_id not in sim_dataset.scene_indices:
+            raise ValueError(f"scene: {scene_id} not in {sim_dataset.scene_indices}")
+
+        self.scene_id = scene_id
+        self.recorded_dataset = sim_dataset.recorded_scene_dataset_batch[scene_id]
+        self.simulated_dataset = sim_dataset.scene_dataset_batch[scene_id]
+
+        self.recorded_ego_states = self.recorded_dataset.dataset.frames
```

Comment:
Would it make sense to add some methods in the recorded_dataset such as `get_frames()` ?

Reply:
right now it's not P0, but I'm happy to go over this later